### PR TITLE
NN-3228 add integration tests

### PR DESCRIPTION
**Pull request by JamieDraperUK**
Initial bit of work to begin converting the Groovy integration tests to Kotlin. This includes about six tests, seven more to go. 

I'm not massively convinced with the Wiremock data strategy for integration tests. It seemed like the cleanest way to keep the previous test data during the conversion.

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class KeyworkerSearchIntegrationTest : IntegrationTest() {
+  companion object {
+    const val PRISON_ID = "LEI"
+    const val USERNAME = "Another"
+  }
+
+  val STAFF_DETAILS = getWiremockResponse(PRISON_ID, "staff-location-role-list")
+  val CASE_NOTE_USAGE = getWiremockResponse("case-note-usage")
+
+  fun `keyworker search - decorated with defaults after migration`() {
```
Comment (JamieDraperUK) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt:12]: Skipping for now. Will pick it back up later.

File: build.gradle.kts [hunk 2]
```diff
@@ -36,7 +36,7 @@ dependencies {
   implementation("com.google.code.gson:gson:2.8.6")
   compileOnly("org.projectlombok:lombok:1.18.18")
   runtimeOnly("org.hsqldb:hsqldb:2.5.1")
-  runtimeOnly("org.flywaydb:flyway-core:7.5.3")
+  implementation("org.flywaydb:flyway-core:7.5.3")
```
Comment (JamieDraperUK) [inline build.gradle.kts:39]: Needed by the IntegrationTest class for cleaning the database after each int test

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt (hunk 1 above)
Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt:12]: You could annotate it with @Disabled with some explanation text - useful as a reminder.

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/health/ComplexityOfNeedQueueHealthIntegrationTest.kt [hunk 3]
```diff
@@ -189,8 +199,14 @@ class ComplexityOfNeedQueueHealthIntegrationTest : IntegrationTest() {
   }
 
   private fun subPing(status: Int) {
-    eliteMockServer.stubFor(
-      WireMock.get("/health/ping").willReturn(
+    addConditionalPingStub(eliteMockServer, status)
```
Comment (JamieDraperUK) [inline src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/health/ComplexityOfNeedQueueHealthIntegrationTest.kt:202]: Some duplication here that will be resolved in future commits.

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt (hunk 1 above)
Comment (JamieDraperUK) [reply src/test/java/uk/gov/justice/digital/hmpps/keyworker/integration/KeyworkerSearchIntegrationTest.kt:12]: I'll add that now locally, thanks Mike

**Pull request merged**